source/service.go

```diff
@@ -33,15 +37,28 @@ type serviceSource struct {
 	namespace string
 	// set to true to process Services with legacy annotations
 	compatibility bool
+	fqdntemplate  *template.Template
 }
 
 // NewServiceSource creates a new serviceSource with the given client and namespace scope.
-func NewServiceSource(client kubernetes.Interface, namespace string, compatibility bool) Source {
+func NewServiceSource(client kubernetes.Interface, namespace string, compatibility bool, fqdntemplate string) (Source, error) {
```

Comment:
Personally I would like to have the signature

    func NewServiceSource(client kubernetes.Interface, namespace, fqdntemplate string, compatibility bool) (Source, error) {